mealy/error_analyzer.py

```diff
@@ -240,7 +242,16 @@ def _compute_primary_model_error(self, X, y):
              error_y: array of string of shape (n_sampled_X, )
              Boolean value of whether or not the primary model predicted correctly or incorrectly the samples in sampled_X.
         """
-        y_pred = self._primary_model.predict(X)
+        if is_regressor(self._primary_model):
+            y_pred = self._primary_model.predict(X)
+        else: # classification model
+            if len(np.unique(y)) > 2: # multiclass -> nothing to do
+                y_pred = self._primary_model.predict(X)
+            else: # binary -> need to check the proba threshold
+                prediction_index = (self._primary_model.predict_proba(X)[:, 1] > self.probability_threshold).astype(int)
+                #map the prediction indexes to the original target values
+                y_pred = [self._primary_model.classes_[i] for i in prediction_index]
```

Comment:
- Nitpick:
```suggestion
        if is_regressor(self._primary_model) or len(np.unique(y)) > 2:
           # Regression or multiclass classification models: no proba threshold
            y_pred = self._primary_model.predict(X)
        else:
            # Binary classification model
            prediction_index = (self._primary_model.predict_proba(X)[:, 1] > self.probability_threshold).astype(int)
            # Map the prediction indexes to the original target values
            y_pred = [self._primary_model.classes_[i] for i in prediction_index]
```

- Also small remark, for binary classif we end uup with a liist, but in other casses we end up with an array-like (depending on X type, can be a np.array). Ensure if this matters

Reply:
@AgatheG which variable are you talking about ? `y_pred` ?

Reply:
Good point, for classif the variable is only used here https://github.com/dataiku/mealy/blob/fix/add-proba-threshold/mealy/error_analyzer.py#L285 , which works between list and array.

Reply:
I casted it to np.array in d9ee24a anw